### PR TITLE
favorite add remove ui / STA-405

### PR DESCRIPTION
**Pull request by MisoMurin**
- drawables added
- star checkbox for adding/removing favorite added to detail activity
- insert success check fixed

### CODE REVIEW

File: app/src/main/java/com/sygic/travel/sdkdemo/detail/PlaceDetailActivity.kt [hunk 1]
```diff
@@ -22,20 +19,57 @@ import com.sygic.travel.sdkdemo.utils.Utils
...
+    private val favoriteAddRemoveCallback = object : Callback<String?>() {
+        override fun onSuccess(data: String?) {}
+
+        override fun onFailure(t: Throwable) {
+	        runOnUiThread { Toast.makeText(this@PlaceDetailActivity, t.message, Toast.LENGTH_LONG).show() }
+	        t.printStackTrace()
+        }
+    }
+
+    private val favoritesCallback = object : Callback<List<String>?>() {
```
Comment (MisoMurin) [inline app/src/main/java/com/sygic/travel/sdkdemo/detail/PlaceDetailActivity.kt]: If you are not satisfied with any naming, feel free to suggest better name :)

File: app/src/main/java/com/sygic/travel/sdkdemo/detail/PlaceDetailActivity.kt [hunk 2]
```diff
@@ -180,35 +214,20 @@ class PlaceDetailActivity : AppCompatActivity() {
...
+        internal var tvDescription: TextView = findViewById(R.id.tv_description) as TextView
+        internal var tvRating: TextView = findViewById(R.id.tv_rating) as TextView
+        internal var tvAddress: TextView = findViewById(R.id.tv_address) as TextView
+        internal var tvPhone: TextView = findViewById(R.id.tv_phone) as TextView
+        internal var tvEmail: TextView = findViewById(R.id.tv_email) as TextView
+        internal var tvAdmission: TextView = findViewById(R.id.tv_admission) as TextView
+        internal var tvOpeningHours: TextView = findViewById(R.id.tv_opening_hours) as TextView
+        internal var fblTags: FlexboxLayout = findViewById(R.id.fbl_tags) as FlexboxLayout
+        internal var llReferencesList: LinearLayout = findViewById(R.id.ll_references_list) as LinearLayout
+        internal var cbFavorite: CheckBox = findViewById(R.id.cb_favorite) as CheckBox
```
Comment (MisoMurin) [inline app/src/main/java/com/sygic/travel/sdkdemo/detail/PlaceDetailActivity.kt]: why? What does "ls" stand for?

File: app/src/main/java/com/sygic/travel/sdkdemo/favorites/FavoritesActivity.kt [hunk 3]
```diff
@@ -28,7 +28,7 @@ class FavoritesActivity : AppCompatActivity() {
 		}
 
 		override fun onFailure(t: Throwable) {
-			Toast.makeText(this@FavoritesActivity, t.message, Toast.LENGTH_LONG).show()
+			runOnUiThread { Toast.makeText(this@FavoritesActivity, t.message, Toast.LENGTH_LONG).show() }
```
Comment (MisoMurin) [inline app/src/main/java/com/sygic/travel/sdkdemo/favorites/FavoritesActivity.kt]: I agree, I usually do that.

File: sdk/src/main/java/com/sygic/travel/sdk/provider/DataProvider.kt [hunk 4]
```diff
@@ -161,7 +161,7 @@ internal class DataProvider(
 	fun addPlaceToFavorites(id: String, back: Callback<String>?) {
 		Thread(Runnable {
 			val insertedCount = stDb?.favoriteDao()?.insert(Favorite(id))
-			if (insertedCount == 1L) {
+			if (insertedCount != null && insertedCount > 0L) {
```
Comment (MisoMurin) [inline sdk/src/main/java/com/sygic/travel/sdk/provider/DataProvider.kt]: The thing is that the method doesn't return a number of inserted rows, but a `rowid` of an inserted row.

File: app/src/main/res/drawable/favorite_selector.xml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+	<item android:state_checked="false" android:drawable="@drawable/ic_favorite_star_off" />
+	<item android:state_checked="true" android:drawable="@drawable/ic_favorite_star_on" />
+</selector>
```
Comment (MisoMurin) [inline app/src/main/res/drawable/favorite_selector.xml]: I've added it, I don't know why it's not here.

File: app/src/main/java/com/sygic/travel/sdkdemo/detail/PlaceDetailActivity.kt [hunk 6]
```diff
@@ -180,35 +214,20 @@ class PlaceDetailActivity : AppCompatActivity() {
...
-            tvRating = findViewById(R.id.tv_rating) as TextView
-            tvAddress = findViewById(R.id.tv_address) as TextView
-            tvPhone = findViewById(R.id.tv_phone) as TextView
-            tvEmail = findViewById(R.id.tv_email) as TextView
-            tvAdmission = findViewById(R.id.tv_admission) as TextView
-            tvOpeningHours = findViewById(R.id.tv_opening_hours) as TextView
-            fblTags = findViewById(R.id.fbl_tags) as FlexboxLayout
-            llReferencesList = findViewById(R.id.ll_references_list) as LinearLayout
-        }
+        internal var ivPhoto: ImageView = findViewById(R.id.iv_detail_photo) as ImageView
```
Comment (MisoMurin) [inline app/src/main/java/com/sygic/travel/sdkdemo/detail/PlaceDetailActivity.kt]: ✔️

File: app/src/main/java/com/sygic/travel/sdkdemo/detail/PlaceDetailActivity.kt [hunk 7]
```diff
@@ -21,12 +18,37 @@ import com.sygic.travel.sdkdemo.gallery.GalleryActivity
...
+
+        override fun onFailure(t: Throwable) {
+	        runOnUiThread { Toast.makeText(this@PlaceDetailActivity, t.message, Toast.LENGTH_LONG).show() }
+	        t.printStackTrace()
+        }
+    }
+
+    private val loadAllFavoritesIdsCallback = object : Callback<List<String>?>() {
+        override fun onSuccess(data: List<String>?) {
+	        checkIfPlaceInFavorites(data)
```
Comment (hrach) [inline app/src/main/java/com/sygic/travel/sdkdemo/detail/PlaceDetailActivity.kt]: still wrong indentation, the best would by format all files by IDE formatter

File: app/src/main/java/com/sygic/travel/sdkdemo/detail/PlaceDetailActivity.kt [hunk 8]
```diff
@@ -21,12 +18,37 @@ import com.sygic.travel.sdkdemo.gallery.GalleryActivity
...
+            setOnFavoriteChangeListener()
+        }
+
+        override fun onFailure(t: Throwable) {
+	        setOnFavoriteChangeListener()
+        }
+    }
+
+	private fun checkIfPlaceInFavorites(data: List<String>?) {
```
Comment (hrach) [inline app/src/main/java/com/sygic/travel/sdkdemo/detail/PlaceDetailActivity.kt]: bad method name. This does not check anything, it sets some property, or inits that property.

File: app/src/main/java/com/sygic/travel/sdkdemo/detail/PlaceDetailActivity.kt (hunk 2 above)
Comment (hrach) [inline app/src/main/java/com/sygic/travel/sdkdemo/detail/PlaceDetailActivity.kt]: well, that's the nevereding discussion if boolean variables should be named with is prefix. I'm ok with leting this as it is.

File: sdk/build.gradle [hunk 9]
```diff
@@ -52,7 +51,7 @@ android {
 dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
 
-    compile "org.jetbrains.kotlin:kotlin-stdlib-jre7:$kotlin_version"
+    compile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
```
Comment (hrach) [inline sdk/build.gradle]: I've read that this may fix some problem, did you encounter any problems?

File: sdk/build.gradle [hunk 10]
```diff
@@ -82,3 +81,4 @@ dependencies {
 apply from: 'maven.gradle'
 apply from: 'dokka.gradle'
 apply from: 'bintray.gradle'
+apply plugin: 'kotlin-android-extensions'
```
Comment (hrach) [inline sdk/build.gradle]: In officials demoves this plugin is at beggining. Why the move?

File: app/src/main/java/com/sygic/travel/sdkdemo/detail/PlaceDetailActivity.kt (hunk 2 above)
Comment (MisoMurin) [inline app/src/main/java/com/sygic/travel/sdkdemo/detail/PlaceDetailActivity.kt]: Oh, that's Is, not Ls. Well, getter for a boolean property is prefixed with "is", so the property itself is not.

File: app/src/main/java/com/sygic/travel/sdkdemo/detail/PlaceDetailActivity.kt (hunk 7 above)
Comment (MisoMurin) [inline app/src/main/java/com/sygic/travel/sdkdemo/detail/PlaceDetailActivity.kt]: yes, but not in this branch. I'd like to finish this part and refactor it in master or better said in new branch from master.

File: sdk/build.gradle (hunk 9 above)
Comment (MisoMurin) [inline sdk/build.gradle]: What exactly should  may fix problems? Removing `jre-7` or adding it?

File: sdk/build.gradle (hunk 10 above)
Comment (MisoMurin) [inline sdk/build.gradle]: I was just trying something and forgot to move it back

File: sdk/build.gradle (hunk 9 above)
Comment (hrach) [inline sdk/build.gradle]: removing should fix something, it didn't help me, though

**Pull request merged**